spikeinterface/sorters/launcher.py

```diff
@@ -46,6 +44,111 @@ def _run_one(arg_list):
+
+    working_folder: str
+        The working directory.
+
+    sorter_params: dict of dict with sorter_name as key
+        This allow to overwrite default params for sorter.
+
+    mode_if_folder_exists: 'raise_if_exists' or 'overwrite' or 'keep'
+        The mode when the subfolder of recording/sorter already exists.
+            * 'raise' : raise error if subfolder exists
+            * 'overwrite' : delete and force recompute
+            * 'keep' : do not compute again if f=subfolder exists and log is OK
+
+    engine: str
+        'loop', 'joblib', or 'dask'
+
+    engine_kwargs: dict
+        This contains kwargs specific to the launcher engine:
+            * 'loop' : no kwargs
+            * 'joblib' : {'n_jobs' : } number of processes
+            * 'dask' : {'client':} the dask client for submiting task
+
+    verbose: bool
+        default True
+
+    docker_image: None or str
+        If str run the sorter inside a container (docker) using the docker package.
+
+    **sorter_params: keyword args
+        Spike sorter specific arguments (they can be retrieved with 'get_default_params(sorter_name_or_class)'
+
+    Returns
+    -------
+
+    sorting : UnitsAggregationSorting
+        The aggregated SortingExtractor.
+
+    Examples
+    --------
+
+    This example shows how to run spike sorting split by group using the 'joblib' backend with 4 jobs for parallel
+    processing.
+
+    >>> sorting = si.run_sorter_by_property("tridesclous", recording, grouping_property="group",
+                                            working_folder="sort_by_group", engine="joblib",
+                                            engine_kwargs={"n_jobs": 4})
+
+    """
+
+    assert grouping_property in recording.get_property_keys(), f"The 'grouping_property' {grouping_property} is not " \
+                                                               f"a recording property!"
+    recording_dict = recording.split_by(grouping_property)
+
+    sorting_output = run_sorters([sorter_name], recording_dict, working_folder,
+                                 mode_if_folder_exists=mode_if_folder_exists,
+                                 engine=engine,
+                                 engine_kwargs=engine_kwargs,
+                                 verbose=verbose,
+                                 with_output=True,
+                                 docker_images={sorter_name: docker_image},
+                                 sorter_params={sorter_name: sorter_params})
+
+    grouping_property_values = np.array([])
+    sorting_list = []
+    for (output_name, sorting) in sorting_output.items():
+        prop_name, sorter_name = output_name
+        sorting_list.append(sorting)
+        grouping_property_values = np.concatenate((grouping_property_values, [prop_name] * len(sorting.get_unit_ids())))
+
+    aggregate_sorting = aggregate_units(sorting_list)
+    aggregate_sorting.set_property(key=grouping_property, values=grouping_property_values)
+
```

Comment:
You shoudl lso handle the delete forlder stuff here no ?

Reply:
No because the `run_sorters` don't have this option. Should we "manually" add it?

Reply:
OK. Leave we will see this later. run_sorters and deleting is not relevant unless we copy sorting to a small format at each step.
Lets do it if really needed.